mat64/format.go

```diff
@@ -9,15 +9,66 @@ import (
 	"strconv"
 )
 
-// Format prints a pretty representation of m to the fs io.Writer. The format character c
+// Formatted returns a fmt.Formatter for the matrix m using the given options.
+func Formatted(m Matrix, options ...FormatOption) fmt.Formatter {
+	f := formatter{
+		matrix: m,
+		dot:    '.',
+	}
+	for _, o := range options {
+		o(&f)
+	}
+	return f
+}
+
+type formatter struct {
+	matrix Matrix
+	prefix string
+	margin int
+	dot    byte
```

Comment:
If we want to do all the things necessary to handle variable width dots then yes. I'd rather not do that.